From 577bc207756116813421cebc8076e28dc475d01b Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 14:42:26 +0000
Subject: [PATCH] filemanager/boxes.c: cleanup -Wcast-qual warning

boxes.c:207:23: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
                       (void *) skin_name, FALSE);
                       ^
boxes.c:218:31: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
                               (void *) skin_name, FALSE);
                               ^
boxes.c:232:54: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
         listbox_get_current (skin_list, &skin_label, (void **) &skin_name);
                                                      ^
--
boxes.c:814:23: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
     sort_orders_names = panel_get_sortable_fields (&sort_names_num);
                       ^
boxes.c:851:17: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     g_strfreev ((gchar **) sort_orders_names);
                 ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 src/filemanager/boxes.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/src/filemanager/boxes.c b/src/filemanager/boxes.c
index 6e004f0..a53fe02 100644
--- a/src/filemanager/boxes.c
+++ b/src/filemanager/boxes.c
@@ -188,7 +188,8 @@ sel_skin_button (WButton * button, int action)
     int result;
     WListbox *skin_list;
     WDialog *skin_dlg;
-    const gchar *skin_name;
+    gchar *skin_name;
+    gchar *skin_name_default = g_strdup ("default");
     int lxx, lyy;
     unsigned int i;
     unsigned int pos = 1;
@@ -202,17 +203,16 @@ sel_skin_button (WButton * button, int action)
                     DLG_COMPACT);
 
     skin_list = listbox_new (1, 1, 11, 22, FALSE, NULL);
-    skin_name = "default";
-    listbox_add_item (skin_list, LISTBOX_APPEND_AT_END, 0, skin_name_to_label (skin_name),
-                      (void *) skin_name, FALSE);
+    listbox_add_item (skin_list, LISTBOX_APPEND_AT_END, 0, skin_name_to_label (skin_name_default),
+                      (void *) skin_name_default, FALSE);
 
-    if (strcmp (skin_name, current_skin_name) == 0)
+    if (strcmp (skin_name_default, current_skin_name) == 0)
         listbox_select_entry (skin_list, 0);
 
     for (i = 0; i < skin_names->len; i++)
     {
         skin_name = g_ptr_array_index (skin_names, i);
-        if (strcmp (skin_name, "default") != 0)
+        if (strcmp (skin_name, skin_name_default) != 0)
         {
             listbox_add_item (skin_list, LISTBOX_APPEND_AT_END, 0, skin_name_to_label (skin_name),
                               (void *) skin_name, FALSE);
@@ -237,6 +237,7 @@ sel_skin_button (WButton * button, int action)
         button_set_text (button, str_fit_to_term (skin_label, 20, J_LEFT_FIT));
     }
     dlg_destroy (skin_dlg);
+    g_free (skin_name_default);
 
     return 0;
 }
@@ -805,7 +806,7 @@ panel_listing_box (WPanel * panel, int num, char **userp, char **minip, int *use
 const panel_field_t *
 sort_box (dir_sort_options_t * op, const panel_field_t * sort_field)
 {
-    const char **sort_orders_names;
+    char **sort_orders_names;
     gsize i;
     gsize sort_names_num = 0;
     int sort_idx = 0;
@@ -824,7 +825,7 @@ sort_box (dir_sort_options_t * op, const panel_field_t * sort_field)
         quick_widget_t quick_widgets[] = {
             /* *INDENT-OFF* */
             QUICK_START_COLUMNS,
-                QUICK_RADIO (sort_names_num, sort_orders_names, &sort_idx, NULL),
+                QUICK_RADIO (sort_names_num, (const char **)(void *) sort_orders_names, &sort_idx, NULL),
             QUICK_NEXT_COLUMN,
                 QUICK_CHECKBOX (N_("Executable &first"), &op->exec_first, NULL),
                 QUICK_CHECKBOX (N_("Cas&e sensitive"), &op->case_sensitive, NULL),
